### ISSUE TITLE
command line lesson (lesson 5) will differ depending on who is compiling the document

### ISSUE THREAD

**Issue by chendaniely** [swcarpentry/r-novice-inflammation#236]
If I `make lesson-rmd`  I will always get a change for the some of the chunks because it calls functions like `sessionInfo()`.  The difference is from my computer being a different OS than the current lesson (which seems was generated from OSX)

Is there a better way to handle this?

This might just be an issue us maintainers need to deal with...

**Comment by katrinleinweber** [comment]
This is bugging me as well now, also because the messages are in German. Makes no sense to commit _those_. I'm ignoring it manually for now.

However, in order to avoid this as an error source, _and_ keep the OS version numbers in the examples up-to-date, one of us should update it before a lesson release?

**Comment by diyadas** [comment]
@fmichonneau : @katrinleinweber and I are resolving a lot of issues today :) What do you think of this one?

**Comment by katrinleinweber** [comment]
Maybe related to #376.

**Comment by fmichonneau** [comment]
I can set up the repo to be built and deployed from Travis so it will always be the same environment that generates the markdown files.

**Comment by diyadas** [comment]
I like that solution. Thanks @fmichonneau !

**Comment by fmichonneau** [comment]
I should be able to set it up by the end of the week. Do you think it will be possible to get the PR merged/closed by then? Enabling Travis requires switching the default branch, and it would affect open PRs.

**Comment by diyadas** [comment]
Oh! @katrinleinweber should confirm - I know she was planning some work during these next two weeks which includes updating and merging one of the open PRs.

**Comment by katrinleinweber** [comment]
Yes, I have nothing planned during September, as I'll be on vacation. As we discussed in the last call, we completed #298 & I verified that we have no headings deeper than a single level per episode. So, go Travis :-)

**Comment by diyadas** [comment]
Hi @fmichonneau - any chance we could make the switch this week? I'd love to go ahead before more PRs roll in.

Thanks!

**Comment by katrinleinweber** [comment]
I'm on holiday until the end of September. Please feel free to go ahead.

**Comment by fmichonneau** [comment]
working on this now. I'll let you when it's completed and functional.

**Comment by fmichonneau** [comment]
The transition to Travis is now complete. A few important considerations:

- **The default branch for this repository is now `master` (this will affect existing pull requests)**.
- After each push to `master` Travis takes the content of the repository, knits the episodes, and push the content of the repository to the `gh-pages` branch. From there GitHub takes care of rendering the website at the usual location.
- The `gh-pages` branch is an orphan branch: any manual commit to `gh-pages` will be erased/lost once Travis rebuilds the lesson.
- I included a check for the links included in the lesson. This is run after the "deploy" stage. At this stage, if there are broken links, the build still passes (the badge will appear green on the README). There are currently some false positive there. I'll try to get back to this in a few days.

A few things for you to consider as maintainers:
- you may want to go through the repository to check for mentions in the documentation about how to contribute to this repository and make sure instructions are updated to reflect the names of the branches and the fact that travis is being used to build and deploy the lesson
- you may consider deleting all the files generated by scripts in the `master` branch (all the files in `_episodes`, the generated figures in `fig/`, ...) this will reduce the probability that someone will make contributions to the wrong file.

Don't hesitate to get in touch with me if you have any questions.
